app/main/views/users.py

```diff
@@ -100,6 +104,9 @@ def create_user():
     json_payload = json_payload["users"]
     validate_user_json_or_400(json_payload)
 
+    if not user_email_domain_allowed(json_payload):
```

Comment:
Should we do this in the model validation instead? Users are unlikely to be created anywhere outside this endpoint, but it still feels like the model attribute validation is a more reliable place for this